web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,281 @@
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.ontologyList = response.graph.ontology;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+        );
+    }]);
+
+mudrodControllers.controller('metadataViewCtrl', ['$rootScope', '$scope', '$routeParams', 'MetaData', 'PagerService', 'SearchOptions', 
+    function metadataViewCtrl($rootScope, $scope, $routeParams, MetaData, PagerService, SearchOptions) {
+        var vm = this;
+        vm.PDItems = [];
+        vm.pager = {};
+        vm.setPage = setPage;
+        vm.totalMatches = 0;
+
+        var word = new String();
+        var opt = new String();
+        
+        if(!$routeParams.query){
+            word = $rootScope.searchOptions.query;    
+            opt = $rootScope.searchOptions.opt; 
+            SearchOptions.setSearchOptions({'query':word, 'opt':opt});
+        } else {
+            word = $routeParams.query;
+            opt = $routeParams.opt;
+            
+            var searchKeyWords = new String();
+
+            if (word.search(',') != -1) {
+                var topics = word.split(',');
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } else {
+                searchKeyWords = word;
+            }
+            
+            if (searchKeyWords.search('/') != -1 ) {
+                var topics = searchKeyWords.split('/');
+                searchKeyWords = "";
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } 
+            
+            word = searchKeyWords;
+
+            if(!$rootScope.searchOptions) {
+                $rootScope.searchOptions = {};
+                opt = 'Or';
+            }
+            else
+                opt = $rootScope.searchOptions.opt;
+
+            SearchOptions.setSearchOptions({'query':word, 'opt':opt});
+            $rootScope.searchOptions.query = searchKeyWords;
+            $rootScope.searchOptions.opt = opt;
+            
+            //opt = 'And';
+        }
+
+        function initController() {
+            // initialize to page 1
+            vm.setPage(1);
+        }
+
+        function setPage(page) {
+            if (page < 1 || page > vm.pager.totalPages) {
+                return;
+            }
+
+            // get pager object from service
+            vm.pager = PagerService.GetPager(vm.PDItems.length, page);
+
+            // get current page of items
+            vm.items = vm.PDItems.slice(vm.pager.startIndex, vm.pager.endIndex + 1);
+        }
+
+        this.searchTopic = function(topic) {
+            var searchKeyWords = new String();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a literal value for this. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3APrimitiveWrappers)